### PR TITLE
Rename GitHub Apps installation and installation_repositories events.

### LINKED ISSUE #627: Support new GitHub Apps (formerly Integrations) API.
GitHub API Announcement:
https://developer.github.com/changes/2017-04-25-Integrations-Pre-Release/

### PR DESCRIPTION
**Pull request by bradleyfalzon**
With the release of GitHub Apps (was GitHub Integrations), GitHub
have renamed the `integration_installation` and `integration_installation_repositories`
event to installation and installation_repositories respectively.

This is a breaking change for those using this API, and can be fixed with:

```bash
gofmt -w -r 'IntegrationInstallationEvent -> InstallationEvent' [path ...]
gofmt -w -r 'IntegrationInstallationRepositoriesEvent -> InstallationRepositoriesEvent' [path ...]
```

Closes #627.
Updates #434.

### CODE REVIEW

**Comment by bradleyfalzon** [comment]
The PR closes 627 as I believe this was the only action item (address the breaking changes).

File: github/event_types.go [hunk 1]
```diff
@@ -187,25 +187,23 @@ type TeamChange struct {
...
-// GitHub API docs: https://developer.github.com/early-access/integrations/webhooks/#integrationinstallationrepositoriesevent
-type IntegrationInstallationRepositoriesEvent struct {
-	// The action that was performed. Possible values for an "integration_installation_repositories"
-	// event are: "added", "removed".
+// GitHub API docs: https://developer.github.com/v3/activity/events/types/#installationrepositoriesevent
+type InstallationRepositoriesEvent struct {
+	// The action that was performed. Can be either "added" or "removed".
 	Action              *string       `json:"action,omitempty"`
 	RepositoriesAdded   []*Repository `json:"repositories_added,omitempty"`
 	RepositoriesRemoved []*Repository `json:"repositories_removed,omitempty"`
```
Comment (shurcooL) [inline github/event_types.go]: This struct is missing `repository_selection` field from https://developer.github.com/v3/activity/events/types/#installationrepositoriesevent. Is it in scope of this PR too add it?

**Comment by shurcooL** [comment]
Question. I don't see any entries for these 2 events in `messages_test.go`. Is there a reason for that, or is it unintentional? Could you add it as part of this PR too?

**Comment by bradleyfalzon** [comment]
I was only addressing the renaming of fields in preparation for the removal, I can address the additional fields and tests too. Once there's agreement on `libraryVersion` I'll make the other changes then.

**Comment by shurcooL** [comment]
I'm completely okay with keeping the scope of this PR with dealing with the breaking API change rename, and leaving the rest out. Up to you.

**Comment by bradleyfalzon** [comment]
Let me know what you want to do with the library version, and I'll take a closer look and just look to PR two commits.

**Comment by shurcooL** [comment]
> Let me know what you want to do with the library version

It's currently "7", it should be bumped up by one to become "8".

File: github/event_types.go [hunk 2]
```diff
@@ -187,25 +187,23 @@ type TeamChange struct {
...
+// InstallationEvent is triggered when a GitHub App has been installed or uninstalled.
+// The Webhook event name is "installation".
 //
-// GitHub API docs: https://developer.github.com/early-access/integrations/webhooks/#integrationinstallationevent
-type IntegrationInstallationEvent struct {
-	// The action that was performed. Possible values for an "integration_installation"
-	// event are: "created", "deleted".
+// GitHub API docs: https://developer.github.com/v3/activity/events/types/#installationevent
+type InstallationEvent struct {
+	// The action that was performed. Can be either "created" or "deleted".
```
Comment (bradleyfalzon) [inline github/event_types.go]: @shurcooL do you want me to remove these comments? They duplicate the GitHub API and it's only a comment for a single field, yet the GitHub API have docs for other fields. So either these should be removed or all of the fields should have the docs duplicated IMHO - is there another reason they are there?

Comment (shurcooL) [inline github/event_types.go]: No, keep them, they're helpful and appreciated (as long as they're accurate, and I've checked, they are).

**Comment by bradleyfalzon** [comment]
OK, leave this with me, I'll get at it within a week.

**Comment by bradleyfalzon** [comment]
OK, I've pushed the changes as discussed in the first commit, `libraryVersion` was incremented, `repository_selection` was added to `InstallationRepositoriesEvent` and tests were added to `messages_test.go`. Can you please confirm these are the changes that were intended?

I've also pushed another commit in the related GitHub Apps service. Essentially, renaming it from `IntegrationsService` to `AppsService`. I don't use this service at all, so hopefully these changes are correct and follow conventions. Specifically should it be `AppsService` or `AppService` (the GitHub docs URLs refer to `/apps/` but the API refers to `/app/`). Also, the file rename from `github/integration_installation.go` to `github/installation.go`, I was unsure whether it should have been renamed to `github/apps_installation.go` to follow the docs hierarchy https://developer.github.com/v3/apps/installations or the API `GET /installation/repositories`.

**Comment by shurcooL** [comment]
> OK, I've pushed the changes as discussed in the first commit, `libraryVersion` was incremented, `repository_selection` was added to `InstallationRepositoriesEvent` and tests were added to `messages_test.go`. Can you please confirm these are the changes that were intended?

Looks good.

> I've also pushed another commit in the related GitHub Apps service. Essentially, renaming it from `IntegrationsService` to `AppsService`. I don't use this service at all, so hopefully these changes are correct and follow conventions. Specifically should it be `AppsService` or `AppService` (the GitHub docs URLs refer to `/apps/` but the API refers to `/app/`). Also, the file rename from `github/integration_installation.go` to `github/installation.go`, I was unsure whether it should have been renamed to `github/apps_installation.go` to follow the docs hierarchy https://developer.github.com/v3/apps/installations or the API `GET /installation/repositories`.

Let's go with `AppsService`. It's consistent with docs, and with the plural versions of `RepositoriesService`, `IssuesService`, etc. Naming the file `apps_installation.go` with the `apps_` prefix SGTM.

**Comment by bradleyfalzon** [comment]
Thanks @shurcooL, I've added the `apps_` prefix. I believe everything is in order if you do.

File: github/apps.go [hunk 3]
```diff
@@ -7,17 +7,17 @@ package github
 
 import "context"
 
-// IntegrationsService provides access to the installation related functions
+// AppsService provides access to the installation related functions
 // in the GitHub API.
 //
-// GitHub API docs: https://developer.github.com/v3/integrations/
-type IntegrationsService service
+// GitHub API docs:	https://developer.github.com/v3/apps/
```
Comment (shurcooL) [inline github/apps.go]: There should be a single space after colon.

File: github/github.go [hunk 4]
```diff
@@ -127,7 +127,7 @@ type Client struct {
 	Gists          *GistsService
 	Git            *GitService
 	Gitignores     *GitignoresService
-	Integrations   *IntegrationsService
+	Apps           *AppsService
```
Comment (shurcooL) [inline github/github.go]: Let's move it up so it's alphabetical.

File: github/github.go [hunk 5]
```diff
@@ -216,7 +216,7 @@ func NewClient(httpClient *http.Client) *Client {
 	c.Gists = (*GistsService)(&c.common)
 	c.Git = (*GitService)(&c.common)
 	c.Gitignores = (*GitignoresService)(&c.common)
-	c.Integrations = (*IntegrationsService)(&c.common)
+	c.Apps = (*AppsService)(&c.common)
```
Comment (shurcooL) [inline github/github.go]: Also here.

**Comment by bradleyfalzon** [comment]
All fixed, can you review again?

**Comment by shurcooL** [comment]
I feel this I should ready from you side, I can merge.

**Comment by bradleyfalzon** [comment]
I'm happy with this. This isn't an extensive update of all GitHub Apps end points, it was just focusing on making the naming changes required from Integrations to GitHub Apps. There's still some missing pieces there.

**Pull request merged**